Description: Patches to use the libobjc2 BlocksRuntime
 This allows the user to provide a version of libBlocksRuntime on
 systems other than Darwin.
 .
 libdispatch (5.5-1) unstable; urgency=low
 .
   * Initial release
Author: Hugo Melder <contact@hugomelder.com>

---
Last-Update: 2023-09-09

--- libdispatch-5.5.orig/CMakeLists.txt
+++ libdispatch-5.5/CMakeLists.txt
@@ -147,6 +147,8 @@ if(NOT CMAKE_SYSTEM_NAME STREQUAL Androi
   find_package(LibRT)
 endif()
 
+find_package(BlocksRuntime QUIET)
+
 check_function_exists(_pthread_workqueue_init HAVE__PTHREAD_WORKQUEUE_INIT)
 check_function_exists(getprogname HAVE_GETPROGNAME)
 check_function_exists(mach_absolute_time HAVE_MACH_ABSOLUTE_TIME)
--- libdispatch-5.5.orig/src/CMakeLists.txt
+++ libdispatch-5.5/src/CMakeLists.txt
@@ -1,5 +1,4 @@
-
-if(NOT CMAKE_SYSTEM_NAME STREQUAL Darwin)
+if(NOT BlocksRuntime_FOUND)
   add_subdirectory(BlocksRuntime)
 endif()
 
--- libdispatch-5.5.orig/tests/CMakeLists.txt
+++ libdispatch-5.5/tests/CMakeLists.txt
@@ -78,9 +78,6 @@ function(add_unit_test name)
     # to reduce probability of test failures due to machine load.
     target_compile_options(${name} PRIVATE -DLENIENT_DEADLINES=1)
   endif()
-  target_include_directories(${name}
-                             SYSTEM BEFORE PRIVATE
-                               "${BlocksRuntime_INCLUDE_DIR}")
   if("${CMAKE_C_SIMULATE_ID}" STREQUAL "MSVC")
     target_compile_options(${name} PRIVATE -Xclang -fblocks)
     target_compile_options(${name} PRIVATE /W3 -Wno-deprecated-declarations)
